mne/viz/_3d.py

```diff
@@ -953,14 +954,134 @@ def _limits_to_control_points(clim, stc_data, colormap):
+    if colormap == 'auto':
+        colormap = mne_analyze_colormap(clim, format='matplotlib')
+
+    time_label, times = _handle_time(time_label, time_unit, stc.times)
+    if initial_time is None:
+        time_idx = 0
+    else:
+        time_idx = np.argmin(np.abs(times - initial_time))
+    fig = plt.figure() if figure is None else figure
+    ax = Axes3D(fig)
+
+    surf = op.join(subjects_dir, subject, 'surf', '%s.%s' % (hemi, surface))
+    if spacing == 'all':
+        coords, faces = nib.freesurfer.read_geometry(surf)
+        inuse = slice(None)
+    else:
+        stype, sval, ico_surf, src_type_str = _check_spacing(spacing)
+        surf = _create_surf_spacing(surf, hemi, subject, stype, ico_surf,
+                                    subjects_dir)
+        inuse = surf['vertno']
+        faces = surf['use_tris']
+        coords = surf['rr'][inuse]
+        shape = faces.shape
+        faces = stats.rankdata(faces, 'dense').reshape(shape) - 1
+
+    hemi_idx = 0 if hemi == 'lh' else 1
+    if hemi_idx == 0:
+        data = stc.data[:len(stc.vertices[0]), time_idx:time_idx + 1]
+    else:
+        data = stc.data[len(stc.vertices[0]):, time_idx:time_idx + 1]
+    vertices = stc.vertices[hemi_idx]
+    n_verts = len(vertices)
+    tris = _get_subject_sphere_tris(subject, subjects_dir)[hemi_idx]
+    e = mesh_edges(tris)
+    e.data[e.data == 2] = 1
+    n_vertices = e.shape[0]
+    maps = sparse.identity(n_vertices).tocsr()
+    e = e + sparse.eye(n_vertices, n_vertices)
+    array_plot = _morph_buffer(data, vertices, e, smoothing_steps, n_verts,
+                               inuse, maps)
+
+    vmax = np.max(array_plot)
+    colors = array_plot / vmax
+    cmap = cm.get_cmap(colormap)
+    greymap = cm.get_cmap('Greys')
+
+    curv = nib.freesurfer.read_morph_data(
+        op.join(subjects_dir, subject, 'surf', '%s.curv' % hemi))[inuse]
+    curv = np.clip(np.array(curv > 0, np.int), 0.2, 0.8)
+
+    transp = 0.8
+
+    polyc = ax.plot_trisurf(*coords.T, triangles=faces, antialiased=False)
+    color_ave = np.zeros(len(faces))
+    curv_ave = np.zeros(len(faces))
+    for idx, face in enumerate(faces):
+        color_ave[idx] = np.mean(colors[face])
+        curv_ave[idx] = np.mean(curv[face])
+    facecolors = art3d.PolyCollection.get_facecolors(polyc)
+
+    to_blend = color_ave > ctrl_pts[0] / vmax
+    facecolors[to_blend, :3] = ((1 - transp) *
+                                greymap(curv_ave[to_blend])[:, :3] +
+                                transp * cmap(color_ave[to_blend])[:, :3])
+    facecolors[~to_blend, :3] = greymap(curv_ave[~to_blend])[:, :3]
+    ax.view_init(**kwargs[views])
+    ax.axis('off')
+    ax.set_title(time_label % times[time_idx], color='w')
+    ax.set_xlim(-80, 80)
+    ax.set_ylim(-80, 80)
+    ax.set_zlim(-80, 80)
+    ax.set_aspect('equal')
+    with warnings.catch_warnings(record=True):  # deprecation warn in mpl >= 2
+        ax.set_axis_bgcolor(background)
```

Comment:
better to do `try: newway() except AttributeError: oldway()` instead of just catching the warning